File: app/src/main/java/com/musicabinet/mobile/ui/home/HomeAdapter.java
package com.musicabinet.mobile.ui.home;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * @author Kirchhoff-
 */

public class HomeAdapter extends PagerAdapter {

    private Context context;

    public HomeAdapter(Context context) {
        this.context = context;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        HomePagerEnum customPagerEnum = HomePagerEnum.values()[position];
        LayoutInflater inflater = LayoutInflater.from(context);
        ViewGroup layout = (ViewGroup) inflater.inflate(customPagerEnum.getLayoutResId(),
                container, false);
        container.addView(layout);
        return layout;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object view) {
        container.removeView((View) view);
    }

    @Override
    public int getCount() {
        return HomePagerEnum.values().length;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        HomePagerEnum customPagerEnum = HomePagerEnum.values()[position];
        return context.getString(customPagerEnum.getTitleResId());
    }
}

File: app/src/main/java/com/musicabinet/mobile/ui/home/HomeAdapter.kt
package com.musicabinet.mobile.ui.home

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.view.View
import android.view.ViewGroup

import com.musicabinet.mobile.R
import com.musicabinet.mobile.ui.home.news.HomeNewsView
import com.musicabinet.mobile.ui.home.tutorial.HomeTutorialView

/**
 * @author Kirchhoff-
 */

class HomeAdapter(private val context: Context) : PagerAdapter() {

    private lateinit var homeNewsView: HomeNewsView
    private lateinit var homeTutorialView: HomeTutorialView

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        return if (position == 0) {
            homeNewsView = HomeNewsView(context)
            container.addView(homeNewsView)
            homeNewsView
        } else {
            homeTutorialView = HomeTutorialView(context)
            container.addView(homeTutorialView)
            homeTutorialView
        }
    }

    override fun destroyItem(container: ViewGroup, position: Int, view: Any) {
        container.removeView(view as View)
    }

    override fun getCount() = 2

    override fun isViewFromObject(view: View, `object`: Any) = view === `object`

    override fun getPageTitle(position: Int): CharSequence? {
        return if (position == 0)
            context.getString(R.string.news_tab)
        else
            context.getString(R.string.tutorial_tab)
    }
}
